test/core.jl

```diff
@@ -745,6 +745,64 @@ let a = []
     @test length(a) == 1
 end
 
+# issue #1784
+let a = []
+    try
+        return 5
+        3
+    finally
+        push!(a, 1)
+    end
+    @test x == 5
```

Comment:
Where does `x` come from? I think the `return` above is causing these tests to be skipped.